minu loogika

karmelp opened the pull request:
Hakkasin seda edit taski tegema.

Kõigepealt tegin edit nupu taski peale.

Siis mötlesin, et teen modali funktsiooni ja siis veel modal content funktsiooni aga mötlen, et äkki peaks isegi veel rohkem seda lahti võtma.

Nimelt. Teeks get close modal button eraldi funktsioonina ja siis see edit task save buttoni ka ja selle sisu jne…

Vöi ma juba ajan seda liiga keeruliseks?

Sest praaeeeguse koodi järgi… ma tegin, et modal display none alguses… ja siis peaks klikkimisel getModal funktsioon avanema, aga siis peaks ka modali display blockiks muutma… ja siis ma jooksin kokku ja hakkasin oma tehtus kahtlema…

Seega… kas mo loogika on õige või peaks muud moodi minema?

karmelp committed:
minu loogika

Hakkasin seda edit taski tegema.

Kõigepealt tegin edit nupu taski peale.

Siis mötlesin, et teen modali funktsiooni ja siis veel modal content funktsiooni aga mötlen, et äkki peaks isegi veel rohkem seda lahti võtma.

Nimelt. Teeks get close modal button eraldi funktsioonina ja siis see edit task save buttoni ka ja selle sisu jne…

Vöi ma juba ajan seda liiga keeruliseks?

Sest praaeeeguse koodi järgi… ma tegin, et modal display none alguses… ja siis peaks klikkimisel getModal funktsioon avanema, aga siis peaks ka modali display blockiks muutma… ja siis ma jooksin kokku ja hakkasin oma tehtus kahtlema…

Seega… kas mo loogika on õige või peaks muud moodi minema?

hhabakuk commented:
Sul on alguse loogika täiesti loogiline - teed edit nupu ja avad sellele klikkides modali. Aga siin ongi jah see keerukus, et kuidas sa avad ja sulged modalit.

Tavaliselt tehakse siin nii, et modal on staatiline - võid selle HTMLi faili kirjutada. Vaikimisi on modalil küljes CSS
```
visibility: hidden;
```

Ja siis kui klikid edit nuppu, siis selle peale paned modalile külge klassi "visible" või "open" vms, kus on küljes CSS
```
visibility: visible;
```

Klasse saad lisada eemaldada   `modal.classList.add('open');` ja `modal.classList.remove('open');`  meetoditega

karmelp commented:
> Sul on alguse loogika täiesti loogiline - teed edit nupu ja avad sellele klikkides modali. Aga siin ongi jah see keerukus, et kuidas sa avad ja sulged modalit.
> 
> Tavaliselt tehakse siin nii, et modal on staatiline - võid selle HTMLi faili kirjutada. Vaikimisi on modalil küljes CSS
> 
> ```
> visibility: hidden;
> ```
> 
> Ja siis kui klikid edit nuppu, siis selle peale paned modalile külge klassi "visible" või "open" vms, kus on küljes CSS
> 
> ```
> visibility: visible;
> ```
> 
> Klasse saad lisada eemaldada `modal.classList.add('open');` ja `modal.classList.remove('open');` meetoditega

Ohh! Ma mötlesin selle peale, aga siis arvasin, et sa raudselt tahad, et ma dünaamiliselt teeks. Väga super! Tegutsen :)

karmelp committed:
try 2

karmelp commented:
> Sul on alguse loogika täiesti loogiline - teed edit nupu ja avad sellele klikkides modali. Aga siin ongi jah see keerukus, et kuidas sa avad ja sulged modalit.
> 
> Tavaliselt tehakse siin nii, et modal on staatiline - võid selle HTMLi faili kirjutada. Vaikimisi on modalil küljes CSS
> 
> ```
> visibility: hidden;
> ```
> 
> Ja siis kui klikid edit nuppu, siis selle peale paned modalile külge klassi "visible" või "open" vms, kus on küljes CSS
> 
> ```
> visibility: visible;
> ```
> 
> Klasse saad lisada eemaldada `modal.classList.add('open');` ja `modal.classList.remove('open');` meetoditega

Milleks need classListid? Ma jooksin kokku. Võid piiluda, mis tegin oma loogikaga. Alguses visibility asemel oli display block & none

hhabakuk commented:
Siin saad mitmel erineval viisil läheneda, otsusta ära lihtsalt kuidas. 

classlist - lähened klassi kaudu ja muudad stiile css failis. Javascriptis lisad ja eemaldad klassi

visibillity - lähened javascripti sees oleva css kaudu otse. Muudad elemendi stiile javascriptis, klassi pole vaja. 

visibility vs display. Otsusta ära kumba kasutad, mõlemat pole vaja kasutada.
https://stackoverflow.com/questions/133051/what-is-the-difference-between-visibilityhidden-and-displaynone

karmelp commented:
> Siin saad mitmel erineval viisil läheneda, otsusta ära lihtsalt kuidas.
> 
> classlist - lähened klassi kaudu ja muudad stiile css failis. Javascriptis lisad ja eemaldad klassi
> 
> visibillity - lähened javascripti sees oleva css kaudu otse. Muudad elemendi stiile javascriptis, klassi pole vaja.
> 
> visibility vs display. Otsusta ära kumba kasutad, mõlemat pole vaja kasutada. https://stackoverflow.com/questions/133051/what-is-the-difference-between-visibilityhidden-and-displaynone

ma juba valisin selle visibility tee aga see ei tööta :D

karmelp commented:
üritasin seda üks ühele ka teha, see ka ei muutu klikitavaks lehel... 
https://daily-dev-tips.com/posts/vanilla-javascript-modal-pop-up/

hhabakuk commented:
Siis hakka debugima. Debuggimine käib tagant poolt ettepoole.

1. Valideeri, kas `editTaskBtnEl.onclick` töötab `getModal` funktsioonis.
```
  editTaskBtnEl.onclick = function () {
    console.log("clicking on edit task button");
  };
  ```
  
  2. Valideeri, kas editTaskBtnElement on õige element `getModal` funktsioonis.
  
  ```
  console.log("editTaskBtnEl", editTaskBtnEl);
  ```
  
  3. Valideeri, kas `getEditBtnElement` funktsioonis saab `editTaskIconEl` endale onclick funktsiooni külge
  ```
    editTaskIconEl.onclick = () => console.log('clicking on edit task icon element');
  ```
  
Proovi leida, mis katki on. FYI, arvuti on loll. Kui sa ütled:

```
var üks = 1
var üks = 2
var üks = 3
```

Siis arvuti teab, et `var üks = 3`, kuigi sina võid arvata, et ilmselgelt on õige, et `var üks  = 1`.

karmelp committed:
ma ei oska... olen mustmiljon erinevat katsetust teinud ja miski ei tööta

karmelp commented:
> 

mai oska seda debugimist...

ja ma ei oska üldse midagi... olen mustmiljon erinevat katsetust teinud ja miski ei toota

Mo ainuke möte hetkel on, et kui seda editTaskIconEl klikates getEditBtnElement funktsiooni sees, tuleb lahti getmodal funktsioon…

Siis peaks getmodal funktsioon avama modalOverlay klassi… aga see ei tööta… 🤷‍♀️ ma ei tea enam… 

Vahet pole kas teen display, visibility vöi classlistiga… olen reaaalselt punkt punktilt teisest koodist maha viksinud köik, a mo oma ikka tööle ei hakka 😂

karmelp committed:
Update script.js

karmelp committed:
Update script.js

karmelp committed:
sdfg

karmelp committed:
Update script.js

hhabakuk commented on script.js:
```diff
@@ -203,12 +259,18 @@ function getTaskListMessage() {
 }
 
 // Kasutaja saab muuta ülesannet.
+function getModal() {
```
Kustuta see ära. See kirjutab üle su eelmist getmodal funktsiooni

karmelp replied:
ma olin juba, et waaaoow, kas töesti!? Aga see ei muutnud olukorda.

Valideeri, kas editTaskBtnEl.onclick töötab getModal funktsioonis.
  editTaskBtnEl.onclick = function () {
    console.log("clicking on edit task button");
  };

Küsin hoopis sellise küsimuse. Kui ma getEditBtnElement() sees panen editTaskIconEl klikk funktsiooni get modal avamisega, siis miks ma pean getModal() sees tegema klikk funktsiooni 



Jöudsin niipalju ära kirjutada ja siis hakkasin mötlema, aga miks see on seal?! Oota. Let me remove it - worked.

hhabakuk replied:
sul getModal funktsiooni üldse ei käivitatagi, sest sul on 2x getModal kirjutatud - javascript kustutab esimese getModali üle teise getModaliga.

1. tee getModal funktsioonis `console.log('getting modal')` -> mitte midagi ei juhtud
2. kustuta getModal funktsioon ära realt 270 -> eelmine `console.log()` läheb tööle

Nüüd sa oled jõudnud kohta, kus tead, et edit nupul vajutamine töötab, getModal funktsioon käivitatakse ning saad hakata uurima, mis toimub getModal funktsiooni sees - miks see ei ava modalit.

Soovitan getModal ümber nimetada `showModalEl` funktsiooniks - see funktsioon ei tagasta sul modalit, nii et nimetus on vale. Kirjuta funktsioon ümber nii, et `showModalEl` teeks ainult ühte asja - näitaks modalit. 

Nüüd uuri, kes peaks käivitama `showModalEl` funktsiooni. 

---

Kui saad modali avanema, siis eraldi probleem on, kuidas modal sulgeda. Modal ei ole dünaamiline erinevalt tasklistist. Seega modali x-nupul võib kohe küljes olla mingi `closeModalEl` funktsioon. `<span onclick="closeModalEl()">Save</span>`


Ühtlasi, usun, et sul läheks lugemine selgemaks, kui alati kasutad samasugust stiili: otsusta, kas teed `el.onclick = () => {}` või `el.addEventlistener('click', () => {})`. Mulle tundub esimene lihtsamini loetav.

karmelp replied:
Jajaa! Selle sain eile ilusti tööle ja naersin jälle enda üle ja hakkasin editTitle’it tegema… Aga see jäi poolikuks, sest uni tuli peale… Ilmselt täna ka ei jõua. Homme ☺️

hhabakuk replied:
okidoki! seal jah keerukus, kuidas teada, millist titlet muutma pead ja kuidas muudad.

Plaan on umbes selline, et modali avanedes pead käivitama funktsiooni, mis võtab endale kaasa konkreetse `taskTitle`'i, muudab `tasks` arrays õiget taski ning lõpuks käivitab `showAllTasks` funktsiooni uuendatud `tasks` arrayga.  `tasks` arrary muutmiseks soovitan kasutada vana head `.map` meetodit.

karmelp replied:
Ma saan aru, et editTask funktsioon peab vötma sisse uue task inputi value.

Ja save buttonit vajutades ta peab käima läbi array, et leida see öige title, mida muuta tuleb uueks ja midal sulgeda.

Kui varasemalt vötsime elemendi attribute’i selleks, et leida see öige object, siis nüüd on nii mitu asja vahepeal, et kuidas see attribute saada? 🙄

Pushisin hetkel ära, kus ma kokku jooksin (katsetadin mitut moodi, aga ei saanud)…

karmelp committed:
poolik

karmelp committed:
editTask

Ma saan aru, et editTask funktsioon peab vötma sisse uue task inputi value.

Ja save buttonit vajutades ta peab käima läbi array, et leida see öige title, mida muuta tuleb uueks ja midal sulgeda.

Kui varasemalt vötsime elemendi attribute’i selleks, et leida see öige object, siis nüüd on nii mitu asja vahepeal, et kuidas see attribute saada? 🙄

Pushisin hetkel ära, kus ma kokku jooksin (katsetadin mitut moodi, aga ei saanud)…

hhabakuk commented:
Soovitan ikkagi `getModal` ümber nimetada showModalEl funktsiooniks - see funktsioon ei tagasta sul modalit, nii et nimetus on vale. Vaata, et showModal ja closeModal teeks mõlemad ainult ühte asja - muudaks modali display stiile.

Nüüd järgi plaani. 

1. Muuda ära klass `editTaskIconEl` -> sa ei muuda ülesande ikooni vaid nimetust. Pakun välja `editTaskTitleBtn`.
2. Muuda ära klass `editTaskSaveBtn`. See nimi ei tähenda midagi - kas nupp  siis editib või seivib. Pakun nimeks `saveNewTaskTitleBtn`

3. `editTaskTitleBtn` elemendil klikkides võiks juhtuda 2 asja: 
   1. modal peaks avanema
   2. modali sees olevale `saveNewTaskTitleBtn` elemendile peaks olema külge pandud `onClick` funktsioon, mis saab kaasa ülesande nimetuse. Selle `onClick` funktsiooni võiksid nimetada nt `onSaveNewTaskTitleBtn`.
   kirjuta funktsioon onEditTaskTitleBtnClick ja pane see `editTaskTitleBtn` elemendi `onclick` attribuudiks.

4. Kirjuta `onSaveNewTaskTitleBtnClick` funkstioon. Funktsioon peaks võtma 2 argumenti - vana taski nimetus ja uus taski nimetus. seejärel peaks funktsioon käima läbi kõik taskid, kui taski vana nimetus on võrdne praeguse nimetusega, tuleb see asendada uue nimetusega. Seejärel võib modali kinni panna ning kuvada uuesti kõik uuendatud taskid.



--- spikker----

funktsioonid showModal ja closeModal

```
function showModal() {
  modalEl.style.display = "flex";
}

function closeModal() {
  modalEl.style.display = "none";
}

NB peab onclicki külge panema ka faili lõpus, seal kus staatilistele elementidele onclicke külge paned
document.querySelector(".modalCloseButton").onclick = closeModal;
```

`editTaskTitleBtn` elemendi onclick funkstioon

```
function onEditTaskTitleBtnClick(event) {
  const oldTaskTitle = event.target.getAttribute("task-title");
  showModal();

  document.querySelector(".saveNewTaskTitleBtn").onclick = () => {
    const editedTaskTitle = document.getElementById("editTaskInput").value;

    onSaveNewTaskTitleBtn(oldTaskTitle, editedTaskTitle);
  };
}
  ```
  
  
 `saveNewTaskTitleBtn` elemendi onclick funktsioon
 
 ```
 function onSaveNewTaskTitleBtnClick(oldTaskTitle, editedTaskTitle) {
  tasks = tasks.map((task) => {
    if (task.title === oldTaskTitle) {
      return { title: editedTaskTitle, isDone: task.isDone };
    } else {
      return task;
    }
  });

  closeModal();
  showAllTasks();
}
 ```

hhabakuk commented:
Ja siin on kogu kood copypastetud, mille kirjutasin ja mis töötab, kui vaja:

```
// Array
let tasks = [
  { title: "Exercise 1", isDone: true },
  { title: "Exercise 2", isDone: false },
];

// Loo funktsioon, mis tagastab status button elemendi
function getStatusBtnElement(task) {
  // loo i element ülesande staatuse nupu ikooni jaoks
  const statusIconEl = document.createElement("i");
  // lisa statusIconEl i elemendile klass
  statusIconEl.className = "fa-solid fa-circle-check button taskStatusButton";
  // lisa statusIconEl i elemendile pealkirja atribuut
  statusIconEl.setAttribute("task-title", task.title);
  // lisa statusIconEl i elemendile onclick funktsioon
  statusIconEl.onclick = () => toggleTaskStatus(statusIconEl);

  // tagasta statusIconEl i element
  return statusIconEl;
}

// Loo funktsioon, mis tagastab delete button elemendi
function getDeleteBtnElement(task) {
  // loo list element ülesande kustutamise nupu jaoks
  const deleteTaskEl = document.createElement("li");
  // loo i element ülesande kustutamise nupu ikooni jaoks
  const deleteTaskIconEl = document.createElement("i");
  deleteTaskIconEl.className = "fa-solid fa-trash button deleteTaskBtn";
  // lisa deleteTaskIconEl i elemendile onclick funktsioon
  deleteTaskIconEl.onclick = () => deleteTask(task);

  deleteTaskEl.appendChild(deleteTaskIconEl);
  // tagasta deleteTaskEl i element
  return deleteTaskEl;
}

// Loo funktsioon, mis tagastab edit button elemendi
function getEditBtnElement(task) {
  // loo list element ülesande kustutamise nupu jaoks
  const editTaskEl = document.createElement("li");
  // loo i element ülesande muutmise nupu ikooni jaoks
  const editTaskTitleBtn = document.createElement("i");
  // lisa editTaskTitleBtn i elemendile klass
  editTaskTitleBtn.className = "fa-solid fa-pen-to-square button editTaskBtn";
  // lisa editTaskTitleBtn i elemendile pealkirja atribuut
  editTaskTitleBtn.setAttribute("task-title", task.title);
  // lisa editTaskTitleBtn i elemendile onclick funktsioon
  editTaskTitleBtn.onclick = onEditTaskTitleBtnClick;

  editTaskEl.appendChild(editTaskTitleBtn);
  // tagasta editTaskEl i element
  return editTaskTitleBtn;
}

function onEditTaskTitleBtnClick(event) {
  const oldTaskTitle = event.target.getAttribute("task-title");
  showModal();

  document.querySelector(".saveNewTaskTitleBtn").onclick = () => {
    const editedTaskTitle = document.getElementById("editTaskInput").value;

    onSaveNewTaskTitleBtn(oldTaskTitle, editedTaskTitle);
  };
}

const modalEl = document.querySelector(".modalOverlay");

function showModal() {
  modalEl.style.display = "flex";
}

function closeModal() {
  modalEl.style.display = "none";
}

function onSaveNewTaskTitleBtn(oldTaskTitle, editedTaskTitle) {
  tasks = tasks.map((task) => {
    if (task.title === oldTaskTitle) {
      return { title: editedTaskTitle, isDone: task.isDone };
    } else {
      return task;
    }
  });

  closeModal();
  showAllTasks();
}

// Loo funktsioon, mis tagastab task title elemendi
function getTaskTitleElement(taskTitle) {
  // loo div element ülesande pealkirja jaoks
  const taskTitleEl = document.createElement("div");
  // lisa taskTitleEl div elemendile pealkirja atribuut
  taskTitleEl.setAttribute("task-title", taskTitle);
  // pane taskTitleEl div elemendi sisse ülesande pealkiri
  taskTitleEl.textContent = taskTitle;

  // tagasta taskTitleEl div element
  return taskTitleEl;
}

// Loo funktsioon, mis paneb tehtuks märgitud taskile teised stiilid
function markTaskAsDone(statusIcon, taskTitle) {
  statusIcon.style.color = "#42b883";
  taskTitle.style.textDecoration = "line-through";
  taskTitle.style.textDecorationThickness = "2px";
}

// Kasutaja saab lugeda kõiki ülesandeid - pealkirja ja staatust
function showTask(task) {
  // võta HTMList ülesannete konteinerelement
  const taskListEl = document.querySelector(".taskList");
  const taskListItemContainerEl = document.createElement("div");
  taskListItemContainerEl.setAttribute("class", "taskListItemContainer");

  // kasuta getStatusBtnElement funktsiooni showTask funkstiooni sees
  const taskStatusEl = getStatusBtnElement(task);
  // kasuta getTaskTitleElement funktsiooni showTask funkstiooni sees
  const taskTitleEl = getTaskTitleElement(task.title);
  // kasuta editTaskEl funktsiooni showTask funkstiooni sees
  const editTaskEl = getEditBtnElement(task);
  // kasuta getDeleteBtnElement funktsiooni showTask funkstiooni sees
  const deleteTaskEl = getDeleteBtnElement(task);
  // loo unordered list element ülesande tööriistade jaoks
  const taskToolsEl = document.createElement("ul");
  // lisa ülesande tööriistade unordered list elemendile klass
  taskToolsEl.className = "taskTools";

  // kui task on tehtud
  if (task.isDone) {
    // kasuta markTaskAsDone funktsiooni
    markTaskAsDone(taskStatusEl, taskTitleEl);
  }

  // pane deleteTaskEl element taskToolsEl elemendi sisse
  taskToolsEl.append(editTaskEl, deleteTaskEl);
  // kasuta el.append(), kui on elemendile vaja panna mitu last
  taskListItemContainerEl.append(taskStatusEl, taskTitleEl, taskToolsEl);
  // kasuta el.appendChild(), kui on elemendile vaja panna üks laps
  taskListEl.appendChild(taskListItemContainerEl);
}

// Kasutaja saab märkida ülesande tehtuks.
function toggleTaskStatus(el) {
  // käi läbi kõik taskid
  tasks.map((task) => {
    // kui taski title on see, millel klikiti...
    if (task.title === el.getAttribute("task-title")) {
      // muuda taski staatus vastupidiseks
      task.isDone = !task.isDone;
    }
  });

  // joonista uuesti kõik tehtud ülesanded
  showNumberOfDoneTasks();
  // joonista uuesti kõik ülesanded
  showAllTasks();
}

// Kasutaja näeb kõiki ülesandeid
function showAllTasks() {
  //tühjenda HTMLis taskide konteiner
  const taskListEl = document.querySelector(".taskList");
  taskListEl.innerHTML = "";

  // kui ülesandeid pole
  if (tasks.length == 0) {
    // käivita getTaskListMessage funktsioon
    getTaskListMessage();
  }

  // joonista HTML element iga taski jaoks
  tasks.forEach((task) => {
    showTask(task);
  });
}

// Kasutaja saab lisada uue ülesande.
function onAddNewTaskBtnClick() {
  // kui newTaskField input on tühi...
  if (document.querySelector(".newTaskField input").value.length == 0) {
    // siis ütle inimesele, et tühjana ei saa ülesannet lisada
    alert("Write a task first!");
  } else {
    // võta newTaskField input väärtus
    const taskTitle = document.getElementById("newTaskInput").value;
    // lisa newTaskField input väärtus arraysse objektina
    tasks.push({ title: taskTitle, isDone: false });
    // joonista uuesti tehtud ülesannete number
    showNumberofAllTasks();
    // joonista uuesti kõik ülesanded
    showAllTasks();
  }
  // tühjenda input newTaskInput field peale uue ülesande lisamist
  document.getElementById("newTaskInput").value = "";
}

// Kasutaja näeb, mitu on ülesandeid kokku.
function showNumberofAllTasks() {
  // võta HTMList allTasksCount klass
  const allTasksCount = document.querySelector(".allTasksCount");
  // sisesta allTasksCount elemendi sisse number
  allTasksCount.textContent = tasks.length;
}

// Kasutaja näeb, mitu ülesannet on tehtud.
function showNumberOfDoneTasks() {
  // võta HTMList doneTasksCount klass
  const doneTasksCount = document.querySelector(".doneTasksCount");
  //for loop, et loendada kokku tehtud ülesanded
  let doneTasksCounter = 0;
  for (let i = 0; i < tasks.length; i++) {
    if (tasks[i].isDone === true) doneTasksCounter++;
  }

  // sisesta doneTasksCount elemendi sisse number
  doneTasksCount.textContent = doneTasksCounter;
}

// Kasutaja saab ülesande kustutada.
function deleteTask(task) {
  const deleteIndex = tasks.indexOf(task);
  tasks.splice(deleteIndex, 1);

  // joonista uuesti kõik ülesanded
  showAllTasks();
  // joonista uuesti mitu ülesannet tehtud on
  showNumberOfDoneTasks();
  // joonista uuesti mitu ülesannet on kokku
  showNumberofAllTasks();
}

// Kasutaja saab kõik ülesanded korraga ära kustutada.
function deleteAllTasks() {
  const deleteIndex = tasks;
  tasks.splice(deleteIndex, tasks.length);

  // joonista uuesti kõik ülesanded
  showAllTasks();
  // joonista uuesti mitu ülesannet tehtud on
  showNumberOfDoneTasks();
  // joonista uuesti mitu ülesannet on kokku
  showNumberofAllTasks();
}

// Teadaanne, kui ülesandeid pole
function getTaskListMessage() {
  // võta HTMList ülesannete konteinerelement
  const taskListEl = document.querySelector(".taskList");
  // loo div element sõnumi jaoks
  const taskListMessageContainerEl = document.createElement("div");
  // lisa taskListMessageContainerEl div elemendile klass
  taskListMessageContainerEl.className = "taskListMessage";
  // loo p element sõnumi pealkirja jaoks
  const taskListMessageTextEl = document.createElement("p");
  taskListMessageTextEl.innerHTML = "Your task list is empty";
  const taskListMessageIconEl = document.createElement("i");
  taskListMessageIconEl.className =
    "fa-solid fa-clipboard-check taskListMessageIcon";

  taskListMessageContainerEl.append(
    taskListMessageTextEl,
    taskListMessageIconEl
  );
  // pane taskListMessageContainerEl element taskListEl elemendi sisse
  taskListEl.appendChild(taskListMessageContainerEl);
}

// vajutades .deleteAllTasksBtn'it aktiveerub deleteAllTasks funktsioon
document.querySelector(".deleteAllTasks").onclick = deleteAllTasks;

// vajutades .addNewTaskButton'it aktiveerub onAddNewTaskBtnClick funktsioon
document.querySelector(".addNewTaskButton").onclick = onAddNewTaskBtnClick;

document.querySelector(".modalCloseButton").onclick = closeModal;

// näita kohe, kui kasutaja äppi saabub, mitu ülesannet nimekirjas on
showNumberofAllTasks();
// näita kohe, kui kasutaja äppi saabub, mitu ülesannet tehtud on
showNumberOfDoneTasks();
showAllTasks();

// Kasutaja saab korraga kustutada köik tehtud ülesanded.
```

karmelp committed:
editTaskTitleBtn works
